runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -62,6 +62,11 @@ public static LogData getEmpty(long address) {
         return logData;
     }
 
+    @Override
+    public EnumMap<IMetadata.LogUnitMetadataType, Object> getMetadataMap() {
+        return ICorfuPayload.enumMapFromBuffer(serializedMetadataMap, LogUnitMetadataType.class);
```

Comment:
Are we going to deserialize a metadataMap every single time when we need it?
It's gonna be a huge performance impact

Reply:
@pankti-m To solve the oom problem, we only need to serialize metadata before it was put into cache, so I think we need both a serialized and unserialized field for metamap, just like `payload` and `data`.